fixtures/configs/smoketest.toml

```diff
@@ -46,14 +46,14 @@ timeout = 20
 retry_wait_time = 2
 
 # Comma-separated list of accepted status codes for valid links.
-accept = [200, 429]
+accept = "200, 429"
```

Comment:
This change would break existing configs. I wonder if we should keep the array syntax here (and use the string syntax for the CLI). There's likely also a way to support both. At the end, it should work as long as the TOML library knows how to handle it and convert it to our internal format, no? Wanna look into that?

Reply:
I can try to make both syntax work. However, I'm not quite sure if it is possible to have a different format in the config compared to the CLI, as both use the same underlying struct... Let me know if you have any insights on this.

Reply:
Yeah, not 100% sure either. I think you can impl `From<String>` and `From<[&str]>` for the struct, but I'm not certain. Something to investigate.

Reply:
At least this works:

```rust
use serde::Deserialize;

#[derive(Deserialize)]
struct Config {
    accept: Accept,
}

// Accept can be either a string or a list of strings
#[derive(Deserialize, Debug, PartialEq)]
#[serde(untagged)]
enum Accept {
    One(String),
    Many(Vec<String>),
}

fn main() {
    let config1: Config = toml::from_str(r"accept = '200, 429'").unwrap();
    assert_eq!(config1.accept, Accept::One("200, 429".to_string()));

    let config2: Config = toml::from_str(r"accept = ['200', '429']").unwrap();
    assert_eq!(
        config2.accept,
        Accept::Many(vec!["200".to_string(), "429".to_string()])
    );
}
```

Reply:
And this also works:

```rust
use serde::{
    de::{self, Deserializer, SeqAccess, Visitor},
    Deserialize,
};
use std::fmt;

#[derive(Deserialize)]
struct Config {
    accept: Accept,
}

#[derive(Debug, PartialEq)]
struct Accept(Vec<String>);

struct AcceptVisitor;

impl<'de> Visitor<'de> for AcceptVisitor {
    type Value = Accept;

    fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
        formatter.write_str("a string or sequence of strings")
    }

    fn visit_str<E: de::Error>(self, value: &str) -> Result<Accept, E> {
        Ok(Accept(
            value.split(',').map(|s| s.trim().to_string()).collect(),
        ))
    }

    fn visit_seq<S: SeqAccess<'de>>(self, mut seq: S) -> Result<Accept, S::Error> {
        let mut vec = Vec::new();
        while let Some(value) = seq.next_element()? {
            vec.push(value);
        }
        Ok(Accept(vec))
    }
}

impl<'de> Deserialize<'de> for Accept {
    fn deserialize<D: Deserializer<'de>>(deserializer: D) -> Result<Accept, D::Error> {
        deserializer.deserialize_any(AcceptVisitor)
    }
}

fn main() {
    let config: Config = toml::from_str(r"accept = '200, 429'").unwrap();
    assert_eq!(
        config.accept,
        Accept(vec!["200".to_string(), "429".to_string()])
    );

    let config: Config = toml::from_str(r"accept = ['200', '429']").unwrap();
    assert_eq!(
        config.accept,
        Accept(vec!["200".to_string(), "429".to_string()])
    );

    let config: Config = toml::from_str(r"accept = ['200..204', '..=429']").unwrap();
    assert_eq!(
        config.accept,
        Accept(vec!["200..204".to_string(), "..=429".to_string()])
    );
}
```

With that, we should be able to support both variants (&mdash; maybe even from the commandline if we want).
Feel free to ask for support in case you get stuck. 🤗

Reply:
> I can try to make both syntax work. However, I'm not quite sure if it is possible to have a different format in the config compared to the CLI, as both use the same underlying struct... Let me know if you have any insights on this.

The cool thing about this is, that once we impl the deserialization for our accepted status codes like above, the CLI params should also "just work", i.e. they will accept both formats, because `clap` also just calls `deserialize` and as long as it returns a valid value, it's happy with that. 😎

Reply:
That's neat. I will implement this!

Reply:
Implemented in 32f654a